internal/pkg/cli/svc_deploy.go

```diff
@@ -558,45 +558,57 @@ func (o *deploySvcOpts) deploySvc(addonsURL string) error {
 	return nil
 }
 
-func validateAliasAndAppVersion(svcName, alias string, app *config.Application, envName string, appVersionGetter versionGetter) error {
-	if alias == "" {
+func validateLBSvcAliasAndAppVersion(svcName string, aliases *manifest.AliasOverride, app *config.Application, envName string, appVersionGetter versionGetter) error {
+	if aliases == nil {
 		return nil
 	}
+	aliasList, err := aliases.ToStringSlice()
+	if err != nil {
+		return fmt.Errorf(`convert 'http.alias' to string slice: %w`, err)
+	}
 	if err := validateAppVersion(app.Name, appVersionGetter); err != nil {
 		logAppVersionOutdatedError(svcName)
 		return err
 	}
-	// Alias should be within either env, app, or root hosted zone.
-	var regEnvHostedZone, regAppHostedZone, regRootHostedZone *regexp.Regexp
-	var err error
-	if regEnvHostedZone, err = regexp.Compile(fmt.Sprintf(`^([^\.]+\.)?%s.%s.%s`, envName, app.Name, app.Domain)); err != nil {
-		return err
-	}
-	if regAppHostedZone, err = regexp.Compile(fmt.Sprintf(`^([^\.]+\.)?%s.%s`, app.Name, app.Domain)); err != nil {
-		return err
-	}
-	if regRootHostedZone, err = regexp.Compile(fmt.Sprintf(`^([^\.]+\.)?%s`, app.Domain)); err != nil {
-		return err
-	}
-	for _, re := range []*regexp.Regexp{regEnvHostedZone, regAppHostedZone, regRootHostedZone} {
-		if re.MatchString(alias) {
-			return nil
+	for _, alias := range aliasList {
+		// Alias should be within either env, app, or root hosted zone.
+		var regEnvHostedZone, regAppHostedZone, regRootHostedZone *regexp.Regexp
+		var err error
+		if regEnvHostedZone, err = regexp.Compile(fmt.Sprintf(`^([^\.]+\.)?%s.%s.%s`, envName, app.Name, app.Domain)); err != nil {
+			return err
 		}
-	}
-	log.Errorf(`%s must match one of the following patterns:
+		if regAppHostedZone, err = regexp.Compile(fmt.Sprintf(`^([^\.]+\.)?%s.%s`, app.Name, app.Domain)); err != nil {
+			return err
+		}
+		if regRootHostedZone, err = regexp.Compile(fmt.Sprintf(`^([^\.]+\.)?%s`, app.Domain)); err != nil {
+			return err
+		}
+		validAlias := false
+		for _, re := range []*regexp.Regexp{regEnvHostedZone, regAppHostedZone, regRootHostedZone} {
+			if re.MatchString(alias) {
+				validAlias = true
+				break
```

Comment:
nit: can we replace this with:
```suggestion
				return nil
```

Reply:
Hmm I don't think so. We need more alias to validate.